integrationtests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -0,0 +1,190 @@
+import io.kroxylicious.proxy.testkafkacluster.KafkaClusterConfig;
+import io.kroxylicious.proxy.testkafkacluster.KafkaClusterFactory;
+
+import static java.net.http.HttpResponse.BodyHandlers.ofString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MetricsIT {
+
+    private TestInfo testInfo;
+
+    @BeforeEach
+    public void beforeEach(TestInfo testInfo) {
+        this.testInfo = testInfo;
+    }
+
+    public static class ConfigHook implements MicrometerConfigurationHook {
+
+        @Override
+        public void configure(MeterRegistry targetRegistry) {
+            new JvmThreadMetrics().bindTo(targetRegistry);
+        }
+    }
+
+    @Test
+    public void shouldOfferPrometheusMetricsScrapeEndpoint() throws Exception {
+        String proxyAddress = "localhost:9192";
+
+        try (var cluster = KafkaClusterFactory.create(KafkaClusterConfig.builder().testInfo(testInfo).build())) {
+            cluster.start();
+
+            String config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers())
+                    .withPrometheusEndpoint().build();
+
+            var proxy = startProxy(config);
+
+            String counter_name = "test_metric_" + Math.abs(new Random().nextLong()) + "_total";
+            Metrics.counter(counter_name).increment();
+            HttpRequest request = HttpRequest.newBuilder(URI.create("http://localhost:9193/metrics")).GET().build();
+            HttpResponse<String> response = HttpClient.newHttpClient().send(request, ofString());
+            assertResponseBodyContainsMeter(response, counter_name, "1.0");
+            HttpRequest notFoundReq = HttpRequest.newBuilder(URI.create("http://localhost:9193/nonexistant")).GET().build();
+            HttpResponse<String> notFoundResp = HttpClient.newHttpClient().send(notFoundReq, ofString());
+            assertEquals(notFoundResp.statusCode(), HttpResponseStatus.NOT_FOUND.code());
+            proxy.shutdown();
+        }
+
+    }
+
+    @Test
+    public void shouldOfferPrometheusMetricsWithNamedBinder() throws Exception {
+        String proxyAddress = "localhost:9192";
+
+        try (var cluster = KafkaClusterFactory.create(KafkaClusterConfig.builder().testInfo(testInfo).build())) {
+            cluster.start();
+
+            String config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers())
+                    .withMicrometerBinder("JvmGcMetrics")
+                    .withPrometheusEndpoint().build();
+
+            var proxy = startProxy(config);
+            HttpRequest request = HttpRequest.newBuilder(URI.create("http://localhost:9193/metrics")).GET().build();
+            HttpResponse<String> response = HttpClient.newHttpClient().send(request, ofString());
+            assertResponseBodyContainsMeter(response, "jvm_gc_memory_allocated_bytes_total");
+            proxy.shutdown();
```

Comment:
Should proxy be shutdown in a finally block?

Or maybe store a reference to it in a field and shutdown in an after all. I'd really prefer to see the proxy injected along with the Kafka cluster but that's for an other day.

Reply:
Moved the shutdown into an `afterEach` (we need a fresh proxy every time because of the varying proxy configuration)